core/relay/adaptor/text-embeddings-inference/adaptor.go

```diff
@@ -0,0 +1,175 @@
+	case mode.Embeddings:
+		req.Header.Set("Content-Type", "application/json")
+	default:
+		return fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+	return nil
+}
+
+func (a *Adaptor) ConvertRequest(meta *meta.Meta, req *http.Request) (string, http.Header, io.Reader, error) {
+	switch meta.Mode {
+	case mode.Rerank:
+		// Read the request body
+		body, err := io.ReadAll(req.Body)
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to read request body: %w", err)
+		}
+
+		// Parse the request body
+		var requestBody map[string]interface{}
+		if err := json.Unmarshal(body, &requestBody); err != nil {
+			return "", nil, nil, fmt.Errorf("failed to parse request body: %w", err)
+		}
+
+		// Create the rerank request, setting model is not supported
+		rerankRequest := map[string]interface{}{
+			// "model": meta.OriginModel,
+			"query": requestBody["query"],
+			"texts": requestBody["documents"],
+		}
+
+		// Convert back to JSON
+		rerankBody, err := json.Marshal(rerankRequest)
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to marshal rerank request: %w", err)
+		}
+
+		return "", req.Header, bytes.NewReader(rerankBody), nil
+
+	case mode.Embeddings:
+		// Handle embeddings request
+		return "", req.Header, req.Body, nil
+
+	default:
+		return "", nil, nil, fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+}
+
+func (a *Adaptor) DoRequest(meta *meta.Meta, c *gin.Context, req *http.Request) (*http.Response, error) {
+	// Get the request URL
+	url, err := a.GetRequestURL(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new request with context
+	newReq, err := http.NewRequestWithContext(req.Context(), req.Method, url, req.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	for k, v := range req.Header {
+		newReq.Header[k] = v
+	}
+
+	// Setup headers
+	if err := a.SetupRequestHeader(meta, c, newReq); err != nil {
+		return nil, err
+	}
+
+	// Convert request if needed
+	_, headers, body, err := a.ConvertRequest(meta, newReq)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update request with converted data
+	for k, v := range headers {
+		newReq.Header[k] = v
+	}
+	if body != nil {
+		newReq.Body = io.NopCloser(body)
+	}
+
+	// Send request using utils
+	return utils.DoRequest(newReq)
+}
+
+func (a *Adaptor) DoResponse(_ *meta.Meta, c *gin.Context, resp *http.Response) (*model.Usage, *relaymodel.ErrorWithStatusCode) {
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, &relaymodel.ErrorWithStatusCode{
+			Error: relaymodel.Error{
+				Message: fmt.Sprintf("failed to read response body: %v", err),
+			},
+			StatusCode: resp.StatusCode,
+		}
+	}
+
+	// Check for error status
+	if resp.StatusCode != http.StatusOK {
+		return nil, &relaymodel.ErrorWithStatusCode{
+			Error: relaymodel.Error{
+				Message: fmt.Sprintf("request failed with status %d: %s", resp.StatusCode, string(body)),
```

Comment:
for error handling, do not return the entire body data. Refer to other adapters for error handling.